op-challenger/game/keccak/merkle/tree_test.go

```diff
@@ -0,0 +1,214 @@
+			name:      "EmptyBinaryMerkleTree",
+			leafCount: 0,
+			rootHash:  common.HexToHash("2733e50f526ec2fa19a22b31e8ed50f23cd1fdf94c9154ed3a7609a2f1ff981f"),
+		},
+		{
+			name:      "SingleLeaf",
+			leafCount: 1,
+			rootHash:  common.HexToHash("b726b462a50ed30781da82e5293b8c9a35279cb6b6cfe8c3e54ece15a4fbd328"),
+		},
+		{
+			name:      "TwoLeaves",
+			leafCount: 2,
+			rootHash:  common.HexToHash("96a031b4cb1dbe70a20d578ec7ab3c43d3ebfa043e64ca517021aaad7a691817"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tree := NewBinaryMerkleTree()
+			for i := 0; i < test.leafCount; i++ {
+				input := ([136]byte)(bytes.Repeat([]byte{0x01}, 136))
+				tree.AddLeaf(types.Leaf{
+					Input:           input,
+					Index:           big.NewInt(int64(i)),
+					StateCommitment: common.Hash{},
+				})
+			}
+			require.Equal(t, test.rootHash, tree.RootHash())
+		})
+	}
+}
+
+func TestBinaryMerkleTree_ProofAtIndex(t *testing.T) {
+	proof := [BinaryMerkleTreeDepth]common.Hash{}
+	for i := 0; i < BinaryMerkleTreeDepth; i++ {
+		proof[i] = common.Hash{}
+	}
+	tests := []struct {
+		name      string
+		leafCount int
+		index     int
+		proof     [BinaryMerkleTreeDepth]common.Hash
+	}{
+		{
+			name:      "EmptyBinaryMerkleTree",
+			leafCount: 0,
+			index:     0,
+			proof:     proof,
+		},
+		{
+			name:      "SingleLeaf",
+			leafCount: 1,
+			index:     0,
+			proof: [BinaryMerkleTreeDepth]common.Hash{
+				common.HexToHash("0000000000000000000000000000000000000000000000000000000000000000"),
+				common.HexToHash("ad3228b676f7d3cd4284a5443f17f1962b36e491b30a40b2405849e597ba5fb5"),
+				common.HexToHash("b4c11951957c6f8f642c4af61cd6b24640fec6dc7fc607ee8206a99e92410d30"),
+				common.HexToHash("21ddb9a356815c3fac1026b6dec5df3124afbadb485c9ba5a3e3398a04b7ba85"),
+				common.HexToHash("e58769b32a1beaf1ea27375a44095a0d1fb664ce2dd358e7fcbfb78c26a19344"),
+				common.HexToHash("0eb01ebfc9ed27500cd4dfc979272d1f0913cc9f66540d7e8005811109e1cf2d"),
+				common.HexToHash("887c22bd8750d34016ac3c66b5ff102dacdd73f6b014e710b51e8022af9a1968"),
+				common.HexToHash("ffd70157e48063fc33c97a050f7f640233bf646cc98d9524c6b92bcf3ab56f83"),
+				common.HexToHash("9867cc5f7f196b93bae1e27e6320742445d290f2263827498b54fec539f756af"),
+				common.HexToHash("cefad4e508c098b9a7e1d8feb19955fb02ba9675585078710969d3440f5054e0"),
+				common.HexToHash("f9dc3e7fe016e050eff260334f18a5d4fe391d82092319f5964f2e2eb7c1c3a5"),
+				common.HexToHash("f8b13a49e282f609c317a833fb8d976d11517c571d1221a265d25af778ecf892"),
+				common.HexToHash("3490c6ceeb450aecdc82e28293031d10c7d73bf85e57bf041a97360aa2c5d99c"),
+				common.HexToHash("c1df82d9c4b87413eae2ef048f94b4d3554cea73d92b0f7af96e0271c691e2bb"),
+				common.HexToHash("5c67add7c6caf302256adedf7ab114da0acfe870d449a3a489f781d659e8becc"),
+				common.HexToHash("da7bce9f4e8618b6bd2f4132ce798cdc7a60e7e1460a7299e3c6342a579626d2"),
+			},
+		},
+		{
+			name:      "ManyLeaves",
+			leafCount: 20,
+			index:     0,
```

Comment:
All the tests are using index 0 - we should make sure we can create proofs for cases where we aren't targeting the left most node.  Need to hit left most, right most (even if the tree isn't actually full), plus a left and right child from in the middle.